aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/DeleteHandler.java

```diff
@@ -0,0 +1,69 @@
+package software.amazon.codeartifact.repository;
+
+import software.amazon.awssdk.awscore.AwsResponse;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.services.codeartifact.CodeartifactClient;
+import software.amazon.awssdk.services.codeartifact.model.DeleteRepositoryRequest;
+import software.amazon.awssdk.services.codeartifact.model.ResourceNotFoundException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class DeleteHandler extends BaseHandlerStd {
+    private Logger logger;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<CodeartifactClient> proxyClient,
+        final Logger logger) {
+
+
+        this.logger = logger;
+
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
+            .then(progress ->
+                proxy.initiate("AWS-CodeArtifact-Repository::Delete", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                    // STEP 2.1 [construct a body of a request]
+                    .translateToServiceRequest(Translator::translateToDeleteRequest)
+                    // STEP 2.2 [make an api call]
+                    .makeServiceCall((awsRequest, client) -> deleteRepository(progress, client, awsRequest))
+                    // STEP 2.3 [stabilize]
+                    .stabilize((awsRequest, awsResponse, client, model, context) -> repositoryIsDeleted(model, proxyClient))
+                    .success());
+    }
+
+    private AwsResponse deleteRepository(
+        ProgressEvent<ResourceModel, CallbackContext>  progress,
+        ProxyClient<CodeartifactClient> client,
+        DeleteRepositoryRequest awsRequest
+    ) {
+        AwsResponse awsResponse = null;
+        String arn = progress.getResourceModel().getRepositoryName();
```

Comment:
Does getRepositoryName return arn? or is the variable misnamed?
Also, you probably only need to make this call in the catch.

Reply:
misnamed, I'll update.